rest/src/main/kotlin/com/gitlab/kordlib/rest/Image.kt

```diff
@@ -18,19 +18,42 @@ class Image private constructor(val data: ByteArray, val format: Format) {
         }
 
     companion object {
-        fun raw(data: ByteArray, format: Format): Image {
-            return Image(data, format)
+        fun raw(data: ByteArray, format: Format, size: Size): Image {
+            return Image(data, format, size)
         }
 
         suspend fun fromUrl(client: HttpClient, url: String): Image = with(Dispatchers.IO) {
             val call = client.request<HttpResponse>(url) { method = HttpMethod.Get }
+            val size = Size.fromResolution(call.request.url.parameters["size"])
```

Comment:
If the URL does not contain a size (which is possible), this function would end up lying about the actual size of the image.

Reply:
We can't guarentee the actual size, discord send us size <= desired size. And when not specified it uses desired size being equal to 128. We've handled the null case in Size.fromResolution().

Reply:
> And when not specified it uses desired size being equal to 128.

This is exactly the problem. If someone downloads a discord emote from a url they will end up with an image of 64x64, yet the size property will be 128x128.

Note that an `Image` is not necessarily a Discord image either, this class can also be used to upload images to Discord, so irregular file image sizes should also be supported.